### PR TITLE
Fix tokensControllers instantiation

### PR DESCRIPTION
**Pull request by danjm**
## **Description**

This PR addresses some issues that were identified by @Gudahtt when reviewing the diff for v11.7.3 (https://github.com/MetaMask/metamask-extension/pull/22354), although the comments were put on the v11.7.2 PR, because what is currently 11.7.3 was once 11.7.2: https://github.com/MetaMask/metamask-extension/pull/22336

Screenshot from 2023-12-21 08-34-37
Screenshot from 2023-12-21 08-34-23
Screenshot from 2023-12-21 08-33-58

## **Manual testing steps**

1. The token testing scenarios should all pass https://github.com/MetaMask/metamask-extension/tree/develop/test/scenarios/4.%20tokens
2. Fiat values for erc20 tokens should be correctly displayed 
3. For the scenarios in 1 and 2 related to autodetection and token display, the tests should also pass after switching networks and switching networks multiple times in short succession 

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by Gudahtt — DISMISSED** [review]
LGTM!

File: app/scripts/metamask-controller.js [hunk 1]
```diff
@@ -545,7 +545,7 @@ export default class MetamaskController extends EventEmitter {
         ),
       onNetworkDidChange: networkControllerMessenger.subscribe.bind(
         networkControllerMessenger,
-        'NetworkController:stateChange',
```
Comment (danjm) [inline app/scripts/metamask-controller.js:548]: @Gudahtt There is a problem with this change. It requires https://github.com/MetaMask/core/pull/3598, which has not been released yet.

Will leaving this as `'NetworkController:stateChange'` be buggy? Or does it just mean that the listener will be called more than it needs to be?

Comment (danjm) [reply app/scripts/metamask-controller.js:548]: Actually, I addressed this here c65def9baf

Comment (Gudahtt) [reply app/scripts/metamask-controller.js:548]: Good catch. We would have released the network controller alongside this had we noticed that.

**Review by Gudahtt — APPROVED** [review]

**Review by bergeron — APPROVED** [review]

**Pull request merged**